native/spark-expr/src/hash_funcs/utils.rs

```diff
@@ -20,7 +20,10 @@
 #[macro_export]
 macro_rules! hash_array {
     ($array_type: ident, $column: ident, $hashes: ident, $hash_method: ident) => {
-        let array = $column.as_any().downcast_ref::<$array_type>().unwrap();
+        let array = $column
+            .as_any()
+            .downcast_ref::<$array_type>()
+            .expect("downcast failed");
```

Comment:
Would it be better to include some extra (debug) information in these messages (like the array and type that failed)?
The unwrap failing is almost always going to be an internal bug.

Reply:
Yes, it would be better. I will update this.

Reply:
Updated.